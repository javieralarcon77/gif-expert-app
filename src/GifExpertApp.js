import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState( defaultCategories );

    // const handleAdd = () => {
    //     setCategories([
    //         ...categories,
    //         'One Piece'
    //     ]);
    // }

    return (
        <>
            <h2>Gif Expert App</h2>
            <AddCategory 
                setCategories={ setCategories }
            />
            <hr />
            {/* <button onClick={ handleAdd } >Agregar</button> */}
            <ol>
                { 
                    categories.map( (cat) => 
                        <GifGrid 
                            key={ cat } 
                            category={ cat } 
                        />
                    )
                }
            </ol>
        </>
    )
}
