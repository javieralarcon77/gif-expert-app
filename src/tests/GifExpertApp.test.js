import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { GifExpertApp } from '../GifExpertApp';

describe('pruebas sobre el componente <GifExpertApp />',()=>{
    
    test('debe de verse correctamente el componente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    })
    

    test('debe de mostrar una lista de categorias', () => {
        const categories = ['One Piece', 'One Punch Man'];

        const wrapper = shallow(<GifExpertApp defaultCategories={ categories }/>);

        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    

})