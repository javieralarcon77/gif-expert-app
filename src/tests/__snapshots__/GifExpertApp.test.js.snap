// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre el componente <GifExpertApp /> debe de verse correctamente el componente 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
